src/settings/settingsmanager.h

```diff
@@ -0,0 +1,170 @@
+   * and reloads any settings that have been changed in the
+   * meantime by another application.
+   *
+   * This function is called automatically from QSettings's
+   * destructor and by the event loop at regular intervals,
+   * so you normally don't need to call it yourself.
+   *
+   * Description copied from: https://doc.qt.io/qt-6/qsettings.html#sync
+   *
+   */
+  void sync() { instance()->m_settings.sync(); };
+
+  /**
+   * @brief Construct a new Setting object and register it.
+   *
+   * If another setting with the same key is already registered, no new setting
+   * is registered and a pointer to the existing setting is returned. This
+   * function will crash in debug mode in case the existing setting has a
+   * different configuration from the new setting.
+   *
+   * @param key A QSettings valid key.
+   * @param defaultValue The default value to be returned when this setting is
+   * not set. Default value will be null if unset.
+   * @param removeWhenReset Whether or not this setting should actually be
+   * removed when `reset` is called. Default is true.
+   * @param sensitiveSetting Whether or not this is a sensitive setting i.e. a
+   * setting that must not be logged in plain text. Default is false.
+   */
+  static Setting* createOrGetSetting(
+      const QString& key,
+      std::function<QVariant()> defaultValue = []() { return QVariant(); },
+      bool removeWhenReset = true, bool sensitiveSetting = false);
+
+  /**
+   * @brief Construct a new SettingGroup object
+   *
+   * @param groupKey The group key will be used as a prefix for the key of all
+   * settings added to the group e.g. groupKey/mySetting.
+   * @param removeWhenReset Whether or not this settings in this group should
+   * actually be removed when `reset` is called.
+   * @param sensitiveSetting Whether or not settings in this group are
+   * sensitive settings i.e. settings that must not be logged in plain text.
+   * @param acceptedKeys When this list has at least one member, only the listed
+   * keys will be allowed for storing under this group. Other keys will be
+   * ignored.
+   */
+  static SettingGroup createSettingGroup(
+      const QString& groupKey, bool removeWhenReset = true,
+      bool sensitiveSetting = false, QStringList acceptedKeys = QStringList());
+
+#ifdef UNIT_TEST
+  /**
+   * @brief Destroys the SettingsManager singleton and clear stores.
+   *
+   * The SettingsManager Singleton exists for the entire lifetime of the
+   * application. The way our tests are set up now, we have most test cases
+   * running under the same application instance. This way SettingsManager
+   * state can leak amongst tests and therefore this function is required.
+   *
+   */
+  static void testCleanup();
+#endif
+
+ private:
+  SettingsManager(QObject* parent);
+
+  void registerSetting(Setting* setting);
+
+ private:
+  QMap<QString, Setting*> m_registeredSettings;
+  QSettings m_settings;
```

Comment:
It doesn't necessarily load them. When you create a SettingGroup yes it loads settings on init. But that is not true for a Setting object. So, if a setting is registered with the SettingsManager, it knows about it, but it may not have any value in the QSettings underlying storage.